piqueserver/run.py

```diff
@@ -92,10 +93,12 @@ def update_geoip(target_dir):
     print('Download Complete')
     print('Unpacking...')
 
-    with gzip.open(zipped_path, 'rb') as gz:
-        d = gz.read()
-        with open(extracted_path, 'wb') as ex:
-            ex.write(d)
+    with tarfile.open(zipped_path, 'r:gz') as tar:
+        non_standard_path = os.path.join(working_directory, tar.next().name)
+        tar.extractall(working_directory)
```

Comment:
I remembered some security issues with the tarfile program and yes: extractall on tar files is unsafe and allows attackers who control the archive (which is also currently not downloaded securely) to overwrite any file on the system. This needs to be changed.

Reply:
I see two possible solutions:

1. Check every compressed file and extract those that are safe
2. Download files via https and verify them with the md5 checksum (also downloaded securely)

In my opinion, the second one is better since it also performs an integrity check. But I don't know if that could be easily attacked.

Reply:
I would just search the archive for the mmdb file and then extract that. The problem isn't with extracting files in general, it's with using extractall.

Reply:
Done